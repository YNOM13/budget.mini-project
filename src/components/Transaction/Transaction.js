import PropTypes from "prop-types";
function Transaction({transaction:{value,date,comment='comment'}}) {
    return(
        <div className="flex justify-center">
            {value > 0
                ? <div className="basis-[500px] bg-green-300 text-black p-2 m-1 rounded flex items-center flex-wrap">
                    <span className="bg-white p-1 rounded m-2">{date}</span> get
                    <span className="bg-green-600 p-1 rounded m-2">{value}$,</span>
                    from <span className="bg-white p-1 rounded m-2">{comment}</span>
                  </div>
                : <div className="basis-[500px] bg-red-400 text-black p-2 m-1 rounded flex items-center flex-wrap">
                    <span className="bg-white p-1 rounded m-2">{date}</span> spend
                    <span className="bg-red-600 p-1 rounded m-2">{value}$,</span>
                    on <span className="bg-white p-1 rounded m-2">{comment}</span>
                  </div>
            }
        </div>

    )
}

Transaction.propTypes = {
    transaction:PropTypes.shape({
        label:PropTypes.string,
        value:PropTypes.string,
        date:PropTypes.string,
        comment:PropTypes.string
    })
}
Transaction.defaultProps = {
    transaction:{
        label:'',
        value:0,
        comment:"",
        date:new Date().toISOString().substring(0,10)
    }
}
export default Transaction