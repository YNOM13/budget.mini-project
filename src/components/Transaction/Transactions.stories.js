import React from 'react';

import Transaction from './Transaction';

export default {
    title: 'Example/Transaction',
    component: Transaction,
    parameters: {
        // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
        layout: 'fullscreen',
    },
};

const Template = (args) => <Transaction {...args} />;

export const Income = Template.bind({});
Income.args = {
    transaction: {
        label:'some income',
        value:23,
        comment:'text',
        date:'today'
    },
};
