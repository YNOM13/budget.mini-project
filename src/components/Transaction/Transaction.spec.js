import React from "react";
import Transaction from "./Transaction";
import { shallow } from "enzyme"
describe('Transaction component', () => {
    let sut
    let props
    beforeEach(()=>{
        props = {
            transaction: {
                value:'0',
                comment:'test',
                date:'01.01.2023',
            }
        }
        sut = shallow(<Transaction {...props}/>)
    })
    it('should how transaction', () => {
        expect(sut).toMatchSnapshot()
    })
})