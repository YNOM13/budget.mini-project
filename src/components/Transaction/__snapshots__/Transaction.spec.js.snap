// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should how transaction 1`] = `
<div
  className="flex justify-center"
>
  <div
    className="basis-[500px] bg-red-400 text-black p-2 m-1 rounded flex items-center flex-wrap"
  >
    <span
      className="bg-white p-1 rounded m-2"
    >
      01.01.2023
    </span>
     spend
    <span
      className="bg-red-600 p-1 rounded m-2"
    >
      0
      $,
    </span>
    on 
    <span
      className="bg-white p-1 rounded m-2"
    >
      test
    </span>
  </div>
</div>
`;
